### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by smira**
Go modules, Cilium CLI, Helm, etc.

### CODE REVIEW

**Review by DmitriyMV — APPROVED** [review]

File: hack/cloud-image-uploader/go.mod [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
+
+toolchain go1.21.1
 
 require (
 	github.com/Azure/azure-sdk-for-go v68.0.0+incompatible
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.7.0
-	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.0
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v5 v5.1.0
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage v1.3.0
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.8.0-beta.1
```
Comment (frezbo) [inline hack/cloud-image-uploader/go.mod:9]: beta? i don;t really care though

**Review by frezbo — APPROVED** [review]

File: hack/cloud-image-uploader/go.mod (hunk 1 above)
Comment (smira) [reply hack/cloud-image-uploader/go.mod:9]: yes, it gets pulled in as a dependency of dependency :shrug:

**Pull request merged**